src/sql/workbench/contrib/executionPlan/browser/azdataGraphView.ts

```diff
@@ -292,37 +299,37 @@ export class AzdataGraphView {
 		if (!node.id.toString().startsWith(`element-`)) {
 			node.id = `element-${node.id}`;
 		}
+
 		diagramNode.id = node.id;
+		diagramNode.icon = node.type;
+		diagramNode.metrics = this.populateProperties(node.properties);
 
-		if (node.type) {
-			diagramNode.icon = node.type;
+		diagramNode.badges = [];
+		for (let i = 0; node.badges && i < node.badges.length; i++) {
+			diagramNode.badges.push(this.getBadgeTypeString(node.badges[i].type));
 		}
 
-		if (node.properties) {
-			diagramNode.metrics = this.populateProperties(node.properties);
-		}
+		diagramNode.edges = this.populateEdges(node.edges);
 
-		if (node.badges) {
-			diagramNode.badges = [];
-			for (let i = 0; i < node.badges.length; i++) {
-				diagramNode.badges.push(this.getBadgeTypeString(node.badges[i].type));
-			}
+		diagramNode.children = [];
+		for (let i = 0; node.children && i < node.children.length; ++i) {
+			diagramNode.children.push(this.populate(node.children[i]));
 		}
 
-		if (node.edges) {
-			diagramNode.edges = this.populateEdges(node.edges);
-		}
+		diagramNode.description = node.description;
+		diagramNode.cost = node.cost;
+		diagramNode.subTreeCost = node.subTreeCost;
+		diagramNode.relativeCost = node.relativeCost;
+		diagramNode.elapsedTimeInMs = node.elapsedTimeInMs;
 
-		if (node.children) {
-			diagramNode.children = [];
-			for (let i = 0; i < node.children.length; ++i) {
-				diagramNode.children.push(this.populate(node.children[i]));
-			}
-		}
+		diagramNode.costMetrics = {
```

Comment:
These things should become very easy if we make this array.

Reply:
Would the array contain objects that resemble the following?
```
[
   { name: "metric-name", displayValue: 'display-value'}, ...
]